chips/nrf52/src/uart.rs

```diff
@@ -401,19 +403,18 @@ impl kernel::hil::uart::UART for Uarte {
         let regs = unsafe { &*self.regs };
 
         // truncate rx_len if necessary
-        let mut length = rx_len;
-        if rx_len > rx_buf.len() {
-            length = rx_buf.len();
-        }
+        let mut truncated_length = core::cmp::min(rx_len, rx_buf.len());
+        truncated_length = core::cmp::min(truncated_length, 255);
         //**** Probably need to check what happens when you want to wait for more data than the
         //     rx_buffer is sized to fit. I don't know why you'd do this but whatever.
 
-        self.rx_remaining_bytes.set(length);
+        self.rx_remaining_bytes.set(truncated_length);
         self.offset.set(0);
         self.rx_buffer.replace(rx_buf);
         self.set_rx_dma_pointer_to_buffer();
 
-        regs.rxd_maxcnt.write(Counter::COUNTER.val(length as u32));
+        regs.rxd_maxcnt
+            .write(Counter::COUNTER.val(truncated_length as u32));
```

Comment:
this should be `truncated_uart_max_length - min(truncated_length, UART_MAX_BUFFER)`

Reply:
I see now- I'll bring my test rig home with me and implement these changes tonight. 

Thank you for your patience, and sorry I didn't do it right the first time! 
DC